nuget/helpers/lib/NuGetUpdater/DotNetPackageCorrelation/Correlator.cs

```diff
@@ -0,0 +1,197 @@
+        var warnings = new List<string>();
+        foreach (var majorVersion in runtimeVersions)
+        {
+            var releasesJsonPath = Path.Combine(_releaseNotesDirectory.FullName, majorVersion.ToString(), "releases.json");
+            if (!File.Exists(releasesJsonPath))
+            {
+                warnings.Add($"Unable to find releases.json file for version {majorVersion}");
+                continue;
+            }
+
+            var releasesJson = await File.ReadAllTextAsync(releasesJsonPath);
+            var releasesFile = JsonSerializer.Deserialize<ReleasesFile>(releasesJson, SerializerOptions)!;
+
+            foreach (var release in releasesFile.Releases)
+            {
+                foreach (var sdk in release.GetSdks())
+                {
+                    if (sdk.Version is null)
+                    {
+                        warnings.Add($"Skipping release with missing version information from {releasesJson}");
+                        continue;
+                    }
+
+                    if (sdk.RuntimeVersion is null)
+                    {
+                        warnings.Add($"Skipping release with missing runtime version information from {releasesJson}");
+                        continue;
+                    }
+
+                    if (runtimePackages.Runtimes.ContainsKey(sdk.RuntimeVersion))
+                    {
+                        // already processed this runtime
+                        continue;
+                    }
+
+                    var packagesAndVersions = new PackageSet();
+                    runtimePackages.Runtimes.Add(sdk.RuntimeVersion, packagesAndVersions);
+                    var runtimeDirectory = new DirectoryInfo(Path.Combine(_releaseNotesDirectory.FullName, majorVersion.ToString(), sdk.RuntimeVersion.ToString()));
+                    var runtimeMarkdownPath = Path.Combine(runtimeDirectory.FullName, $"{sdk.RuntimeVersion}.md");
+                    if (!File.Exists(runtimeMarkdownPath))
+                    {
+                        warnings.Add($"Unable to find expected markdown file {runtimeMarkdownPath}");
+                        continue;
+                    }
+
+                    var markdownContent = await File.ReadAllTextAsync(runtimeMarkdownPath);
+                    var packages = GetPackagesFromMarkdown(runtimeMarkdownPath, markdownContent, warnings);
+                    foreach (var (packageName, packageVersion) in packages)
+                    {
+                        packagesAndVersions.Packages[packageName] = packageVersion;
+                    }
+                }
+            }
+        }
+
+        return (runtimePackages, warnings);
+    }
+
+    public static ImmutableArray<(string Name, SemVersion Version)> GetPackagesFromMarkdown(string markdownPath, string markdownContent, List<string> warnings)
+    {
+        var lines = markdownContent.Split("\n").Select(l => l.Trim()).ToArray();
```

Comment:
Should we trim out `\r` while we're at it just in case?

Reply:
`Trim()` will take care of `\r`.